src/loader.js

```diff
@@ -5,155 +5,198 @@ let Promise = require('promise');
 let _ = require('lodash');
 let TopoSort = require('topo-sort');
 
-// We use this closure to create a .create() method which
-// returns a graphViz file for a given component directory
-function generateGraph(dir, vComp) {
-  let tsort = new TopoSort();
-
-  for (let x of _.keys(dir)) {
-    tsort.add(x, dir[x].requires || []);
+/** Check if a value is a flat value or a component definition */
+let isComponent = (def) => {
+  // Check that it's an object
+  if (typeof(def) !== 'object' && def !== null && def !== undefined) {
```

Comment:
my preference is that we check for explicitly supported types (string, number, function) and then return false if we find one.  I know it's an edge case, but what if we get NaN, we're going to get an error somewhere later on.  We could get NaN if we try to divide two configuration values.